### PR TITLE
Added tests for ProxiedSasLogger

### PR DESCRIPTION
**Pull request by oa-metaswitch**
Test for bifrost_httpclient pull request in cpp-common.

### CODE REVIEW

File: src/ut/httpstack_test.cpp [hunk 1]
```diff
@@ -341,6 +356,61 @@ TEST_F(HttpStackTest, SASCorrelationHeader)
...
+  MockSASMessage* message = mock_sas_find_event(SASEvent::RX_HTTP_REQ);
+  EXPECT_TRUE(message != NULL);
+  EXPECT_EQ(message->var_params[0], "12.34.56.78");
+  EXPECT_EQ(message->static_params[0], 4242);
+
+  stop_stack();
+  mock_sas_collect_messages(false);
+}
+
+// Check that the ProxiedSasLogger picks logs real IPs if not X-Real-Ip header is present
```
Comment (eleanor-merry) [inline src/ut/httpstack_test.cpp]: *Check that the ProxiedSasLogger logs the real IPs if the X-Real-Ip header is not present

**Review by eleanor-merry — APPROVED** [review]
Sure (so long as this passes coverage and valgrind checks when you run make full_test)

**Comment by oa-metaswitch** [comment]
Whoops, it didn't check one error case. I've added a test case for it in the same vein and it now passes all tests.

File: src/ut/httpstack_test.cpp [hunk 2]
```diff
@@ -341,6 +356,119 @@ TEST_F(HttpStackTest, SASCorrelationHeader)
...
+  MockSASMessage* message = mock_sas_find_event(SASEvent::RX_HTTP_REQ);
+  EXPECT_TRUE(message != NULL);
+  EXPECT_EQ(message->var_params[0], "12.34.56.78");
+  EXPECT_EQ(message->static_params[0], 4242);
+
+  stop_stack();
+  mock_sas_collect_messages(false);
+}
+
+// Check that the ProxiedSasLogger picks logs real IPs if X-Real-Ip header is missing
```
Comment (eleanor-merry) [inline src/ut/httpstack_test.cpp]: picks logs? This still isn't quite right.

**Review by eleanor-merry — APPROVED** [review]
LGTM - one comment about comments inline

**Pull request merged**